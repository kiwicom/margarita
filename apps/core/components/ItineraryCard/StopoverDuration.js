// @flow

import * as React from 'react';
import * as DateFNS from 'date-fns';

import StopoverDurationWrapper from './StopoverDurationWrapper';

type Props = {|
  +stopoverDuration: ?number,
  +locationName: ?string,
|};

const getDuration = stopoverDuration => {
  const durationDateFrom = new Date(0, 0, 0, 0, 0);
  const durationDateTo = new Date(0, 0, 0, 0, stopoverDuration);
  return DateFNS.distanceInWords(durationDateFrom, durationDateTo);
};

export default function StopoverDuration({
  stopoverDuration,
  locationName,
}: Props) {
  if (stopoverDuration == null) {
    return null;
  }

  return (
    <StopoverDurationWrapper>
      Stays {getDuration(stopoverDuration)}
      {locationName && ` in ${locationName}`}
    </StopoverDurationWrapper>
  );
}
