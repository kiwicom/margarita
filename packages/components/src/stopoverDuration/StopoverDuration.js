// @flow

import * as React from 'react';
import * as DateFNS from 'date-fns';

type Props = {|
  +stopoverDuration: number,
  +locationName: ?string,
|};

const getDuration = stopoverDuration => {
  const durationDateFrom = new Date(0, 0, 0, 0, 0);
  const durationDateTo = new Date(0, 0, 0, 0, stopoverDuration);
  return DateFNS.distanceInWords(durationDateFrom, durationDateTo);
};

export default function StopoverDuration({
  stopoverDuration,
  locationName,
}: Props) {
  return (
    <>
      Stays {getDuration(stopoverDuration)}
      {locationName && ` in ${locationName}`}
    </>
  );
}
