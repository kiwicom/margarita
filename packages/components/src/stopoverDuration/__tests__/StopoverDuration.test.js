// @flow

import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import StopoverDuration from '../StopoverDuration';

const renderer = new ShallowRenderer();

it('works without location', () => {
  expect(
    renderer.render(
      <StopoverDuration stopoverDuration={100} locationName={undefined} />,
    ),
  ).toMatchSnapshot();
  expect(
    renderer.render(
      <StopoverDuration stopoverDuration={100} locationName={null} />,
    ),
  ).toMatchSnapshot();
});
