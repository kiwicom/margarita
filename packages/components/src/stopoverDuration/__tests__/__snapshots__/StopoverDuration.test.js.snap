// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works without location 1`] = `
<React.Fragment>
  Stays 
  about 2 hours
</React.Fragment>
`;

exports[`works without location 2`] = `
<React.Fragment>
  Stays 
  about 2 hours
</React.Fragment>
`;
